oak/server/rust/oak_runtime/src/node/wasm/mod.rs

```diff
@@ -718,74 +716,73 @@ impl wasmi::ModuleImportResolver for WasiStub {
     }
 }
 
+fn validate_entrypoint(module: &wasmi::Module, entrypoint: &str) -> Result<(), OakStatus> {
+    let abi = WasmInterface::new("validate_entrypoint", None);
```

Comment:
This seems quite hacky... Is there a nicer way that the validation logic can be encapsulated in something that can be reused more ergonomically? If not now, perhaps add a TODO?

Reply:
I thought about an alternative that was less hacky but I thought would involve more boilerplate/copied-stuff.  However, now I come to try it out, it's not too bad &ndash; it adds a new Wasm host function resolver just for validation (which therefore doesn't need any state), and by pushing the body of the `WasmInterface::resolv_func` method into a standalone function, the copy-pasta I was worried about doesn't happen. WDYT?